src/preferences/PreferencesBase.js

```diff
@@ -189,7 +198,13 @@ define(function (require, exports, module) {
                         try {
                             result.resolve(JSON.parse(text));
                         } catch (e) {
-                            result.reject(new ParsingError("Invalid JSON settings at " + path + "(" + e.toString() + ")"));
+                            if (createIfNew) {
+                                // JSON parsing error -- start from scratch
+                                console.log("Invalid JSON settings at " + path + "(" + e.toString() + ") -- contents reset");
+                                result.resolve({});
```

Comment:
@rroshan1 

> How about creating a backup file "state.json.backup" in the same directory and then deleting all text from the "state.json" file.

That's essentially what my suggestion does. With my suggestion, the backup file is in the Trash/Recycle so you don't need to worry about creating unique names for duplicate items or with the user needing to know that files are building up and to delete them. Also, Brackets automatically creates a new "state.json", so no need to manipulate the contents of that file.